src/backend/commands/tablespace.c

```diff
@@ -1736,12 +1750,25 @@ PrepareTempTablespaces(void)
 		tblSpcs[numSpcs++] = curoid;
 	}
 
-	SetTempTablespaces(tblSpcs, numSpcs);
+	setTablespacesFunc(tblSpcs, numSpcs);
 
 	pfree(rawname);
 	list_free(namelist);
 }
 
+/*
+ * PrepareTempTablespaces -- prepare to use temp tablespaces for tables and
+ * files.  No work if already done in current transaction.
+ */
+void
+PrepareTempTablespaces(void)
```

Comment:
Dunno should we init `temp_file_tablespaces` while calling this from `GetDefaultTablespace()`.

Reply:
I found it a little confusing to have two separate functions for preparation. Do you want `PrepareTempTablespaces()` to be separated into two functions for tables and files each?

Reply:
You need to check how often we do unnecessary job by initializing tablespaces which are not currently used. If there any significant cases, then functions should be separated.

But the current approach is enough for MVP.

Reply:
I have checked the usage of `PrepareTempTablespaces()`, and noticed that every call to it was made to ensure that tablespaces from `temp_tablespaces` are set up for file creation.

The only usage that isn't for files is:
https://github.com/arenadata/gpdb/blob/2f3e93faeeb035f0847104a67b112711f71ed7d9/src/backend/commands/tablespace.c#L1488-L1493

It's used only when we are creating a temporary table, so I don't think it's significant.